### PR TITLE
Allow projects outside github.com

### PR DESCRIPTION
**Pull request by saravanan30erd**
Fix: #210 

New Projects feature is shipped to GHES 3.8 now. We need to support non-github.com URLs in this action to make it work in GHES.

We have two options on regex, either allow all supported scheme with matching all FQDN OR make regex more robust by matching only what required as @iansan5653 suggested here. https://github.com/actions/add-to-project/pull/210#discussion_r972309192

I updated regex using second option.

### CODE REVIEW

**Comment by omerbensaadon** [comment]
hey @saravanan30erd this is to enable GHES support right?

File: src/add-to-project.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export async function addToProject(): Promise<void> {
 
   if (!urlMatch) {
     throw new Error(
-      `Invalid project URL: ${projectUrl}. Project URL should match the format https://github.com/<orgs-or-users>/<ownerName>/projects/<projectNumber>`,
+      `Invalid project URL: ${projectUrl}. Project URL should match the format <Github server FQDN>/<orgs-or-users>/<ownerName>/projects/<projectNumber>`,
```
Comment (iansan5653) [inline src/add-to-project.ts:83]:
```suggestion
      `Invalid project URL: ${projectUrl}. Project URL should match the format <GitHub server domain name>/<orgs-or-users>/<ownerName>/projects/<projectNumber>`,
```
Just to make this a little easier to understand - not everyone knows what an FQDN is.

**Review by iansan5653 — COMMENTED** [review]
Overall this looks like a good change. Just two minor suggestions and it should be good to go.

**Comment by omerbensaadon** [comment]
@saravanan30erd thanks for the contribution and congrats on your first public contribution on GitHub 🥳

**Comment by saravanan30erd** [comment]
>> this is to enable GHES support right?

@omerbensaadon Yes.

@iansan5653 updated the changes based on feedback.

**Comment by iansan5653** [comment]
Looking at the failing tests:

- `addToProject › throws an error when url isn't a valid project url`: Looks like we just need to update the error text assertion.
- `addToProject › throws an error when url isn't under the github.com domain`: Looks like we can invert this test: make it "<ins>does not</ins> throw an error when url isn't under the github.com domain" and change the assertion

**Comment by saravanan30erd** [comment]
@iansan5653 updated one test and dist. could you please check this test failure `throws an error when url isn't under the github.com domain`?

**Comment by iansan5653** [comment]
Hi @saravanan30erd - looks like I don't have permission to push directly to your branch so instead I opened a pull request from my fork: https://github.com/saravanan30erd/add-to-project/pull/1. If you merge that PR, it will update this PR automatically and then we can get these changes merged.

**Comment by saravanan30erd** [comment]
@iansan5653 thanks, I merged it.

**Review by iansan5653 — APPROVED** [review]

**Review by iansan5653 — APPROVED** [review]

**Comment by iansan5653** [comment]
Thanks again for the contribution! Published as [`v0.5.0`](https://github.com/actions/add-to-project/releases/tag/v0.5.0) so you should be able to use it now.

**Pull request merged**